lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -657,6 +668,7 @@ def generate_ca(
             ),
             critical=False,
         )
+        .add_extension(key_usage, critical=False)
```

Comment:
Why `critical=False`? Admittedly, I just skimmed the issue and the RFC, but it this part did stand out to me:

```
   This extension MUST appear in certificates that contain public keys
   that are used to validate digital signatures on other public key
   certificates or CRLs.  When this extension appears, it SHOULD be
   marked critical.
```

I'm missing a bit of context here, so maybe this doesn't apply in this case.

Reply:
Good catch, fixed.